core/runtime/impl/core_impl.cpp

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "runtime/impl/core_impl.hpp"
+
+#include "primitives/scale_codec.hpp"
+#include "runtime/impl/runtime_external_interface.hpp"
+#include "runtime/impl/wasm_memory_impl.hpp"
+#include "runtime/impl/wasm_memory_stream.hpp"
+
+namespace kagome::runtime {
+
+  CoreImpl::CoreImpl(common::Buffer state_code,
+                     std::shared_ptr<extensions::Extension> extension,
+                     std::shared_ptr<primitives::ScaleCodec> codec)
+      : state_code_(std::move(state_code)),
+        memory_(extension->memory()),
+        executor_(std::move(extension)),
+        codec_(std::move(codec)) {}
+
+  outcome::result<primitives::Version> CoreImpl::version() {
+    uint64_t version_long =
+        executor_
+            .call(state_code_.toVector(), "Core_version",
+                  wasm::LiteralList({wasm::Literal(0), wasm::Literal(0)}))
+            .geti64();
+
+    // first 32 bits are address and second are length (length not needed)
```

Comment:
And I would also like to state another argument for validation of the length: if the returned length is inconsistent with the actual length, it's better to know it, because it is erroneous behavior which may indicate some serious bug.

Reply:
To validate this we need to encode result back to scale and get its length. Something can go wrong here only if did not implement scale codec properly. Scale codec is tested separately, also there will be cross implementation tests, so that we will be 100% sure that did everything right